src/domain/pricing/bulk-editor/index.tsx

```diff
@@ -0,0 +1,1052 @@
+
+  const activateField = (cellKey: string) => {
+    setActiveFields({ ...activeFields, [cellKey]: true })
+  }
+
+  const deactivateField = (cellKey: string) => {
+    const tmp = { ...activeFields }
+    delete tmp[cellKey]
+
+    setActiveFields(tmp)
+  }
+
+  const toggleActive = (cellKey: string) => {
+    if (activeFields[cellKey]) {
+      deactivateField(cellKey)
+    } else {
+      activateField(cellKey)
+    }
+  }
+
+  const setActiveCellsFromPointers = () => {
+    const next = {}
+    const [variants, regions] = matrix
+
+    for (let i = pointers.minV; i <= pointers.maxV; i++) {
+      const v = variants[i]
+      for (let j = pointers.minR; j <= pointers.maxR; j++) {
+        const r = regions[j]
+        next[getPriceKey(v, r)] = true
+      }
+    }
+
+    setActiveFields(next)
+  }
+
+  /* ********** SELECTORS ********** */
+
+  const isActive = (variantId: string, regionId: string) =>
+    activeFields[getPriceKey(variantId, regionId)]
+
+  const getPriceChange = (variantId: string, regionId: string) => {
+    const key = getPriceKey(variantId, regionId)
+
+    if (key in priceChanges && !priceChanges[key]) {
+      return null
+    }
+    return priceChanges[key]
+  }
+
+  /* ********** HANDLERS ********** */
+
+  /*
+   * A handler that implement logic for mouse click multi select.
+   * Responsible for triggering focus/blur events on an input filed.
+   */
+  const onPriceInputClick = (
+    e: React.MouseEvent<HTMLInputElement>,
+    variantId: string,
+    regionId: string
+  ) => {
+    const cellKey = getPriceKey(variantId, regionId)
+
+    if (e.shiftKey) {
+      e.preventDefault() // do not focus
+
+      const active = isActive(variantId, regionId)
+      toggleActive(cellKey)
+
+      if (active) {
+        const wasFocused = e.target === document.activeElement
+        e.target.blur()
+
+        if (wasFocused) {
+          // find another active cell to focus
+          const id = Object.keys(activeFields).find((k) => k !== cellKey)
+          document.querySelectorAll(`[data-id="${id}"]`)[0].focus()
+        }
+      } else {
+        e.target.focus()
+      }
+    } else {
+      // TODO: the issue here is that `onAmount` is called by the underlying library on blur (which is triggered after this on click handler)
+      // SCENARIO: one cell is focused and the user clicks on an other cell (without shift)
+      // the `onClick` handler of the second cell is called first which executes the following 2 lines and resets the state
+      // after that, onBlur of the first cell, the underlying library calls `onAmount` change which sets last edit value in the state again.
+
+      setCurrentEditAmount(undefined)
+      setActiveFields({ [cellKey]: true })
+    }
+  }
+
+  /*
+   * A handler that implement logic for arrow keys multi select.
+   */
+  const onKeyDown = (
+    e: React.KeyboardEvent<HTMLInputElement>,
+    variantId: string,
+    regionId: string
+  ) => {
+    if (e.key === "Tab") {
```

Comment:
I would suggest to put all the keys in an Enum somewhere to have a quick look at the supported key bindings and to avoid any typo for any usage across the app